kfs-core/src/main/java/edu/arizona/kfs/pdp/batch/AchPayeeBankAcctInputFileType.java

```diff
@@ -0,0 +1,934 @@
+		
+		return fileName;
+	}
+
+	@Override
+	public Object parse(byte[] fileByteContent) throws ParseException {
+		LOG.debug("parse() - start=" + new Date());
+		
+		List<PayeeACHAccount> retval = new ArrayList<PayeeACHAccount>();
+		List<PayeeReportLine> reportLines = new ArrayList<PayeeReportLine>();
+		String systemSource = KFSConstants.EMPTY_STRING;
+		BufferedReader bufferedFileReader = null;
+		
+		try {
+			bufferedFileReader = new BufferedReader(new InputStreamReader(new ByteArrayInputStream(fileByteContent)));
+			
+			loadOverridePayeeMemberSet();
+			
+			List<PayeeACHAccount> payeeAccounts = new ArrayList<PayeeACHAccount>();
+			
+			String fileLine = null;
+			
+			while ((fileLine = bufferedFileReader.readLine()) != null) {
+				
+				if (StringUtils.isBlank(systemSource)) {
+					systemSource = fileLine.substring(PdpConstants.ACHFilePropertyPositions.START_OF_SYSTEM_SOURCE, PdpConstants.ACHFilePropertyPositions.END_OF_SYSTEM_SOURCE).trim();
+				}
+				
+				payeeAccounts.add(PayeeACHAcctFlatFileConverter.convert(fileLine));
+			}
+			
+			LOG.debug(payeeAccounts.size() + " payee accounts loaded");
+			
+			loadExistingRoutingNumberSet(payeeAccounts);
+			
+			Map<String, Person> payees = loadPayees(payeeAccounts);
+			
+			LOG.debug(payees.size() + " payees loaded");
+			
+			for (PayeeACHAccount payeeAccount : payeeAccounts) {
+				processPayeeAcct(getPayee(payeeAccount, payees), payeeAccount, reportLines, retval);
+			}
+			
+			LOG.debug(reportLines.size() + " report lines generated");
+			
+			writeReport(reportLines, systemSource);
+		}
+		catch (IOException e) { 
+			LOG.error("Error encountered reading from file content", e);
+			throw new ParseException("Error encountered reading from file content", e);
+		}
+		finally {
+			try {
+				bufferedFileReader.close();
+			}
+			catch (Exception ex) {
+				LOG.error("Error encountered cloasing BufferedReader");
+			}
+		}
+		
+		LOG.debug("parse() - ends =" + new Date());
+		
+		return retval;
+	}
+
+	@Override
+	public boolean validate(Object parsedFileContents) {
+		return true;
+	}
+
+	@Override
+	public void process(String fileName, Object parsedFileContents) {
+		// empty method so compiler doesn't complain
+
+	}
+
+	@Override
+	public String getAuthorPrincipalName(File file) {
+		String[] fileNameParts = StringUtils.split(file.getName(), "_");
+		if (fileNameParts.length >= 2) {
+			return fileNameParts[1];
+		}
+		
+		return null;
+	}
+
+	@Override
+	public String getTitleKey() {
+		return KFSKeyConstants.MESSAGE_BATCH_UPLOAD_TITLE_ACH_PAYEE;
+	}
+	
+	private Person getPayee(PayeeACHAccount payeeAccount, Map<String, Person> payees) {
+		Person retval = payees.get(getPayeeKey(payeeAccount));
+		
+		if (retval == null) {
+			retval = getPersonFromPayeeInfo(payeeAccount.getPayeeIdNumber(), payeeAccount.getPayeeIdentifierTypeCode());
+		}
+		
+		return retval;
+	}
+	
+	private File getFile(String dir, String prefix, String ext, boolean wantTimestamp) {
```

Comment:
The input wantTimestamp is always true from its sole caller on line 731; unless there are plans to use this helper again in a future ticket, we might want to consider nixing `boolean wantTimestamp`. I'll leave this one to your discretion.